zkevm/tests/aggregation_tests.rs

```diff
@@ -53,41 +63,65 @@ fn test_aggregation_api() {
     };
     log::info!("loaded parameters for degrees {} and {}", k, k_agg);
 
-    let mut prover = Prover::from_params_and_seed(params_inner, params_outer.clone(), seed);
+    let mut prover = Prover::from_params_and_seed(params_inner, params_outer, seed);
+    prover.debug_dir = output_dir.clone();
 
     log::info!("build prover");
 
     //
-    // 2. convert block traces into inner circuit proofs, a.k.a. SNARKs
+    // 2. read inner circuit proofs (a.k.a. SNARKs) from previous dumped file or
+    //    convert block traces into
     //
     let super_circuit_proof = prover
-        .create_target_circuit_proof_batch::<SuperCircuit>(block_traces.as_ref(), &mut rng)
-        .unwrap();
+        .read_target_circuit_proof_from_file::<SuperCircuit>()
+        .unwrap()
+        .unwrap_or_else(|| {
+            log::info!("build super circuit from block traces");
+            prover
+                .create_target_circuit_proof_batch::<SuperCircuit>(block_traces.as_ref(), &mut rng)
+                .unwrap()
+        });
 
-    log::info!("build super circuit from block traces");
+    log::info!("got super circuit proof");
 
     // sanity check: the inner proof is correct
 
     // 3. build an aggregation circuit proof
-    let agg_circuit =
-        AggregationCircuit::new(&params_outer, [super_circuit_proof.snark.clone()], &mut rng);
-    let pk_outer = gen_pk(&params_outer, &agg_circuit, None);
-
-    let instances = agg_circuit.instances();
-    let proof = gen_evm_proof_shplonk(
-        &params_outer,
-        &pk_outer,
-        agg_circuit.clone(),
-        instances.clone(),
+    let agg_circuit = AggregationCircuit::new(
+        &prover.agg_params,
+        [super_circuit_proof.snark.clone()],
         &mut rng,
     );
+
+    let proved_block_count = super_circuit_proof.num_of_proved_blocks;
+    let proof = prover
+        .create_agg_proof_by_agg_circuit(&agg_circuit, &mut rng, proved_block_count)
+        .unwrap();
     log::info!("finished aggregation generation");
 
     // 4. generate bytecode for evm to verify aggregation circuit proof
-    let deployment_code = prover.create_evm_verifier_bytecode(&agg_circuit, pk_outer.get_vk());
+    let agg_vk = prover.agg_pk.as_ref().unwrap().get_vk();
+    let deployment_code = prover.create_evm_verifier_bytecode(&agg_circuit, agg_vk);
     log::info!("finished byte code generation");
 
     // 5. validate the proof with evm bytecode
-    Verifier::evm_verify(deployment_code, instances, proof);
+    Verifier::evm_verify(
+        deployment_code,
+        agg_circuit.instances(),
+        proof.proof.clone(),
+    );
```

Comment:
need also to dump the instance and proof used by Verifier::evm_verify

Reply:
Fix to dump inner-circuit-proof, agg-circuit-proof, vk, instance, and deployment bytecode in commit [478d1d4](https://github.com/scroll-tech/scroll-zkevm/pull/132/commits/478d1d46fe4825b5a91184c6355061251b57c48a).

Will check function `create_evm_verifier_bytecode` impl, then refactor to EVM verifier. Thanks.